packages/bff/src/services/selfcareService.ts

```diff
@@ -1,16 +1,33 @@
 /* eslint-disable @typescript-eslint/explicit-function-return-type */
 
-import { UserId } from "pagopa-interop-models";
+import { TenantId, UserId } from "pagopa-interop-models";
 import {
+  bffApi,
   selfcareV2ClientApi,
   SelfcareV2InstitutionClient,
 } from "pagopa-interop-api-clients";
-import { Logger } from "pagopa-interop-commons";
-import { userNotFound } from "../model/errors.js";
+import { WithLogger, Logger } from "pagopa-interop-commons";
+import { missingSelfcareId, userNotFound } from "../model/errors.js";
+import { TenantProcessClient } from "../clients/clientsProvider.js";
+import { BffAppContext } from "../utilities/context.js";
+import { toApiSelfcareUser } from "../api/selfcareApiConverter.js";
 
 export function selfcareServiceBuilder(
-  selfcareV2Client: SelfcareV2InstitutionClient
+  selfcareV2Client: SelfcareV2InstitutionClient,
+  tenantProcessClient: TenantProcessClient
 ) {
+  const filterByUserNameOfSurname = (
+    users: selfcareV2ClientApi.UserResource[],
+    query?: string
+  ): selfcareV2ClientApi.UserResource[] =>
+    !query
+      ? users
+      : users.filter(
+          (user) =>
+            user.name?.toLowerCase().includes(query.toLowerCase()) ||
+            user.surname?.toLowerCase().includes(query.toLowerCase())
```

Comment:
`UserResource` has a `familyName` property, not a `surname` property, and also scala seems to filter by familyName, could you double check?

Reply:
Sorry, ignore the comment, I realized it's because our model conversion is done after the filter, so we still have surname and not familyName 👌